hw/battery/src/battery.c

```diff
@@ -25,7 +25,12 @@
 #include <battery/battery_prop.h>
 #include <battery/battery_drv.h>
 
+
+#if MYNEWT_VAL(BATTERY_MAX_COUNT)
```

Comment:
The way to go in that case is to create syscfg definition in **hw/battery/syscfg.yml**.
Without it package uses value that is never defined.
something like that:
```yml
    BATTERY_MAX_COUNT:
        description: Maximum number of batteries.
        value: 1
```
Then it would be also possible to omit
```c
#if MYNEWT_VAL(BATTERY_MAX_COUNT)
```
... and just leave it like this:
```c
#define BATTERY_MAX_COUNT MYNEWT_VAL(BATTERY_MAX_COUNT)
```

Reply:
Agree that is the right way! 
I was following the style you used here:
https://github.com/apache/mynewt-core/blob/f5832e09b0679ce14032850f2d797dc088dbb9fd/hw/battery/include/battery/battery.h#L38-L42

Reply:
You are quite right, BATTERY_DRIVERS_MAX is also missing from syscfg.yml.
Here is pull request #2384 that fixes that.

If you still want to contribute your change with syscfg definition please don't hesitate to use force push to your branch so we can limit number of commits with names like `fixes review comments`, and we can merge you commits without squashing them first (that can drop authorship).